src/generators/BaseGenerator.js

```diff
@@ -0,0 +1,98 @@
+import fs from 'fs';
+import handlebars from 'handlebars';
+import mkdirp from 'mkdirp';
+import { sprintf } from 'sprintf-js';
+import urlapi from 'url';
+
+export default class {
+  templates = {};
+
+  constructor({hydraPrefix, templateDirectory}) {
+    this.hydraPrefix = hydraPrefix;
+    this.templateDirectory = templateDirectory;
+
+    this.registerTemplates('', ['_entrypoint.js']);
+  }
+
+  registerTemplates(basePath, paths) {
+    for (let path of paths) {
+      this.templates[path] = handlebars.compile(fs.readFileSync(`${this.templateDirectory}/${basePath}${path}`).toString());
+    }
+  }
+
+  createDir(dir, warn = true) {
+    if (!fs.existsSync(dir)) {
```

Comment:
This api is deprecated, saw it when working on typescript generator. Maybe we could put a polyfill (for example: `@soyuka/exists-sync`)

Reply:
No, `exists` is deprecated but not `existsSync`, it's stated explicitly in the documentation: https://nodejs.org/api/fs.html#fs_fs_existssync_path

> Note that fs.exists() is deprecated, but fs.existsSync() is not.

Reply:
Wokay mb